bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -549,24 +549,24 @@ void ReplicaImp::onMessage<preprocessor::PreProcessResultMsg>(std::unique_ptr<pr
 
 template <>
 void ReplicaImp::onMessage<ReplicaAsksToLeaveViewMsg>(std::unique_ptr<ReplicaAsksToLeaveViewMsg> message) {
-  auto *msg = message.release();
   if (activeExecutions_ > 0) {
-    pushDeferredMessage(msg);
+    pushDeferredMessage(message.release());
```

Comment:
Deffered mechanism has a task, but I hope that you had it does..

Reply:
Yes, I created that task, but haven't implement it yet because we push to deferredMessages_ other messages types, which most of them were not converted to smart pointers yet. I believe it would be more efficient to convert deferredMessages_ to hold smart pointers after we finish converting other messages to smart (all onMessage tasks).